tensorpipe/test/channel/channel_test.h

```diff
@@ -76,14 +74,22 @@ class DataWrapper<tensorpipe::CudaBuffer> {
         stream_(other.stream_) {
     other.cudaPtr_ = nullptr;
     other.length_ = 0;
+    other.stream_ = cudaStreamDefault;
+  }
+
+  DataWrapper& operator=(DataWrapper&& other) {
+    cudaPtr_ = other.cudaPtr_;
+    length_ = other.length_;
+    stream_ = other.stream_;
+    other.cudaPtr_ = nullptr;
+    other.length_ = 0;
+    other.stream_ = cudaStreamDefault;
```

Comment:
Sorry to keep nit-picking, but this isn't a valid move assignment operator: if `this` was holding a resource, this operator is "forgetting" that resource without freeing it properly. One "trick" for a valid move constructor is to `std::swap(...)` all fields. This way, the object that we're moving from will end up destroying the resource. (Note that a similar trick also works for the move constructor, although it's not necessary there because `this` can't possibly be holding a resource yet)